src/app/pages/incident-details-view/incident-details-view.component.ts

```diff
@@ -0,0 +1,532 @@
+      },
+      error: (err) => {
+        console.error(`An error occured when deleting incident ${this.incidentId}`);
+        console.error(err);
+        this.toastr.error("An error occured while processing your request.", "API Error");
+        this.waitState = WS_NONE;
+      }
+    });
+  }
+
+  private clearPending(): void {
+    this.pendingImpacts.clear();
+    this.pendingUpdates.clear();
+    this.impactsToAdd = [];
+    this.updatesToAdd = [];
+    this.impactsToDelete.clear();
+    this.updatesToDelete.clear();
+  }
+
+  resetStartDate(): void {
+    this.startDate = incidentDateToUi(this.incidentCopy.beganAt);
+  }
+
+  resetEndDate(): void {
+    this.endDate = incidentDateToUi(this.incidentCopy.endedAt);
+  }
+
+  // Differs from resetEndDate by also calling runChecks. This is meant to be
+  // used by the UI to remove the end date and mark the incident as still ongoing.
+  // Running the checks is necessary to make sure any errors relating to the end
+  // date are being cleared and we might not be in a state to properly do this when
+  // calling resetEndDate in the init method, henceforth this additional method.
+  removeEndDate(): void {
+    this.resetEndDate();
+    this.runChecks();
+  }
+
+  addNewImpact(impact: Impact): void {
+    if (!impact.reference) {
+      console.error("An Impact needs to reference a component");
+      return;
+    }
+    if (this.pendingImpacts.has(impact.reference)) {
+      console.error(`Attempt to add impact with the reference ${impact.reference} a second time`);
+      return;
+    }
+    this.pendingImpacts.add(impact.reference);
+    this.impactsToAdd.push(impact);
+    this.runChecks();
+  }
+
+  addNewUpdate(update: IncidentUpdateResponseData): void {
+    if (this.pendingUpdates.has(update.order)) {
+      console.error(`Attempt to add update with order ${update.order} a second time`);
+      return;
+    }
+    this.pendingUpdates.add(update.order);
+    this.updatesToAdd.push(update);
+    this.latestUpdateOrder++;
+  }
+
+  markImpactForDeletion(id?: ComponentId): void {
+    if (!id) {
+      console.error("Received a call to markImpactForDeletion without an id");
+      return;
+    }
+    if (this.pendingImpacts.has(id)) {
+      this.pendingImpacts.delete(id);
+      let index: number = 0;
+      for (; index < this.impactsToAdd.length; index++) {
+        if (this.impactsToAdd[index].reference === id) {
+          break;
+        }
+      }
+      this.impactsToAdd.splice(index, 1);
+    } else {
+      this.impactsToDelete.add(id);
+    }
+    this.checkError(incidentAffects);
+  }
+
+  unmarkImpactForDeletion(id?: ComponentId): void {
+    if (!id) {
+      console.error("Received a call to unmarkImpactForDeletion without an id");
+      return;
+    }
+    this.impactsToDelete.delete(id);
+    this.checkError(incidentAffects);
+  }
+
+  markUpdateForDeletion(order: number): void {
+    // TODO What about updating the order property of all remaining pending updates?
```

Comment:
The order should not be changed. The combination of the incident id and the order is the private key of an update.

And for example:

4 < 6 even with no 5 in between, so the order is still the same ;)